token/js/src/extensions/tokenMetadata/actions.ts

```diff
@@ -0,0 +1,133 @@
+import type { ConfirmOptions, Connection, PublicKey, Signer, TransactionSignature } from '@solana/web3.js';
+import type { Field } from '@solana/spl-token-metadata';
+import {
+    createInitializeInstruction,
+    createEmitInstruction,
+    createUpdateFieldInstruction,
+} from '@solana/spl-token-metadata';
+import { sendAndConfirmTransaction, Transaction } from '@solana/web3.js';
+
+import { TOKEN_2022_PROGRAM_ID } from '../../constants.js';
+import { getSigners } from '../../actions/internal.js';
+
+/**
+ * Initializes a TLV entry with the basic token-metadata fields.
+ *
+ * @param connection       Connection to use
+ * @param payer            Payer of the transaction fees
+ * @param updateAuthority  Update Authority
+ * @param mint             Mint Account
+ * @param mintAuthority    Mint Authority
+ * @param name             Longer name of token
+ * @param symbol           Shortened symbol of token
+ * @param uri              URI pointing to more metadata (image, video, etc)
+ * @param multiSigners     Signing accounts if `authority` is a multisig
+ * @param confirmOptions   Options for confirming the transaction
+ * @param programId        SPL Token program account
+ *
+ * @return Signature of the confirmed transaction
+ */
+export async function initializeTokenMetadata(
+    connection: Connection,
+    payer: Signer,
+    updateAuthority: PublicKey,
+    mint: PublicKey,
+    mintAuthority: PublicKey | Signer,
+    name: string,
+    symbol: string,
+    uri: string,
+    multiSigners: Signer[] = [],
+    confirmOptions?: ConfirmOptions,
+    programId = TOKEN_2022_PROGRAM_ID
+): Promise<TransactionSignature> {
+    const [mintAuthorityPublicKey, signers] = getSigners(mintAuthority, multiSigners);
+
+    const transaction = new Transaction().add(
+        createInitializeInstruction({
+            programId,
+            metadata: mint,
```

Comment:
Is it correct that we don't give users the option to have a separate metadata account? Understand for token-2022, the metadata can only live inside the mint, so we abstract that away from users

Reply:
No, you can have metadata in another account, but our CLI isn't going to do it for you.

In terms of metadata, this CLI is only handling Token2022 metadata (in mints).

Reply:
Okay - so the approach of not allowing users to pass in a separate metadata account in this action is correct?

Reply:
Yes!